pilot/pkg/model/validation.go

```diff
@@ -501,6 +501,11 @@ func validateTLSOptions(tls *networking.Server_TLSOptions) (errs error) {
 		// no tls config at all is valid
 		return
 	}
+	if (tls.Mode == networking.Server_TLSOptions_SIMPLE || tls.Mode == networking.Server_TLSOptions_MUTUAL) &&
+		tls.CredentialName != "" {
+		// If tls mode is SIMPLE or MUTUAL, and credential name is specified, credentials are fetched remotely.
```

Comment:
Please revert this. This is effectively creating a one-Of behavior. You also dont know whether the proxy that applies this gateway is an old or new proxy. Per earlier discussions, we decided that we will require users to specify credentialName, in a strictly additive fashion (in addition to specifying certs).